### DISCUSSION TITLE
Basic webserver for displaying last workflow runs and worker status

### DISCUSSION THREAD

**Discussion by matBonet** [taskiq-python/taskiq#269] [Ideas]
Hello, I just recently found this repo and I have to say it looks really good, the docs are top notch and very well structured and it looks perfect for the heavy IO focus of our project. I have two main questions as of now:

1. Is there a way to define task workflows (such as DAGs with rejoining paths), in a language agnostic description such as yaml or json?
2. Is there an existing or easily portable front-end for monitoring workflow execution status, history of past runs and workflows, and worker status?

If there aren't and there's enough interest I could see if I get any traction within my team, so any pointers on how to approach querying this information would be great (we have not chosen a broker as of now, if that matters).

Anyway, here are some examples of what I'm trying to describe:

- [Celery Director](https://ovh.github.io/celery-director/)
![image](https://github.com/taskiq-python/taskiq/assets/13615724/3da6cdc8-e5c1-4c24-87a3-ac093fe66beb)

- [Apache Airflow](https://airflow.apache.org/docs/)
![image](https://github.com/taskiq-python/taskiq/assets/13615724/fce3cdd9-b095-4307-93ce-7ab1810ace0e)

**Comment by s3rius** [comment]
1. Yes there is a way. It's called pipelining. Here's the library for that: https://github.com/taskiq-python/taskiq-pipelines
2. There's no available frontend currently. You can get this information from the context of execution, because pipelines inforation is stored in task's labels.
